### ISSUE TITLE
worker queue is full!

### ISSUE THREAD

**Issue by flexdeviser** [hazelcast/hazelcast#11417]
Currently I am using hz#3.8.3, and topology in my env is:

2 nodes-cluster(provides configurations of my application), so this guy are connected by 13 connections.

12 nodes-cluster(my application workers) these "12 nodes" creates connects to (2 nodes-cluster) to get configurations

1 client(ui) talk to (2 nodes-cluster and 12 nodes-cluster) with connections

The problem:
```
# 2017-09-21 06:50:43 [WARN ] [ClientListenerService:66] - hz.client_0 [dev] [3.8.3]  event clientMessage could not be handled
java.util.concurrent.RejectedExecutionException: Task: com.hazelcast.client.spi.impl.listener.ClientListenerServiceImpl$ClientEventProcessor@77555276 is rejected, the worker queue is full!
```

After running 1~2 days I got the error above and I can't reconnect the (2 nodes-cluster) without restart that cluster.

Does someone can help me to figure out what happened to my application?

**Comment by pveentjer** [comment]
Are you using nearcache, collection events, topics etc? I don't know why the client can't reconnect, but on the client side there are too many unprocessed events and to prevent overload, events are getting dropped when the high watermark has been reached.

**Comment by ahmetmircik** [comment]
@flexdeviser?

**Comment by flexdeviser** [comment]
Sorry guys, I really have few maps with event listener, and all these maps being updated every 30seconds. I'm not sure this is the problem or not, so I created a new testing environment to verify this issue. Please leave this issue here, and give me more time to do my test. Thanks.

**Comment by ahmetmircik** [comment]
waiting to hear from you, thanks for the heads up

**Comment by Donnerbart** [comment]
Maybe your listener implementation is too slow, e.g. blocking to wait on something. Then the event queue cannot be processed. We have an internal diagnostics system, which has an `EventQueuePlugin` coming with Hazelcast 3.9. It collects samples from the event queues, so you can see which kind of events are piling up.

If you run a testing environment, maybe you can try out our latest `3.9-SNAPSHOT` version and activate diagnostics with that plugin. Please see the general documentation here: http://docs.hazelcast.org/docs/latest-development/manual/html/Management/Diagnostics.html

The `EventQueuePlugin` is not enabled by default. You'll have to add another system property to activate it:
```java
-Dhazelcast.diagnostics.enabled=true
-Dhazelcast.diagnostics.metric.level=info
-Dhazelcast.diagnostics.event.queue.period.seconds=30
```

**Comment by flexdeviser** [comment]
Hi guys, 
 
@Donnerbart told me "Maybe your listener implementations are too slow" yesterday, so I created a testing env(as asme as my production env) today.   

12 hz nodes(hz-server) updating 2 maps every 2 seconds, and I also create 2 listeners(entry add update remove) are waiting for changes in my client(hz-client). Every time I got the message I'll post the "EntryEvent" object into "EventBus"(guava), then I can get this messages from event bus with listeners(eventbus). This the whole procedure of my testing env. 

I made a little bit change with hz-client configuration:
clientConfig.setProperty("hazelcast.client.event.queue.capacity", "100");
and also 2 seconds delay in event bus listeners(handler).
this changes let me represent the issue with warn message more easier:
`12:07:56,876  WARN || - [ClientListenerService] hz.hz.client_0.IO.thread-in-0 - hz.client_0 [dev] [3.9-EA] Event clientMessage could not be handled
java.util.concurrent.RejectedExecutionException: Task: com.hazelcast.client.spi.impl.listener.ClientListenerServiceImpl$ClientEventProcessor@631c53d8 is rejected, the worker queue is full!
	at com.hazelcast.util.executor.StripedExecutor$Worker.schedule(StripedExecutor.java:190) ~[hazelcast-3.9-EA.jar:3.9-EA]
	at com.hazelcast.util.executor.StripedExecutor$Worker.access$300(StripedExecutor.java:157) ~[hazelcast-3.9-EA.jar:3.9-EA]`

Ok, thanks for all of your help and I will try to reduce the number of maps updating and optimize my client listener implementations. 

One more question, if I shutdown the client application, does the worker queue destroyed as well?

Thanks
